apps/web/components/I18nLanguageHandler.tsx

```diff
@@ -5,16 +5,26 @@ import { useEffect } from "react";
 import { trpc } from "@calcom/trpc/react";
 
 export function useViewerI18n() {
-  return trpc.viewer.public.i18n.useQuery(undefined, {
+  const { data } = trpc.viewer.public.locale.useQuery(undefined, {
     staleTime: Infinity,
-    /**
-     * i18n should never be clubbed with other queries, so that it's caching can be managed independently.
-     * We intend to not cache i18n query
-     **/
     trpc: {
       context: { skipBatch: true },
     },
   });
+  return trpc.viewer.public.i18n.useQuery(
```

Comment:
This is a waterfall now and thus would be slow to fetch translations. Can we avoid this?
Just a suggestion, maybe we can store the locale in session(and update it there when user changes the locale). 
The benefit being it's readily available along with the session.